packages/react-components/src/templates/ManuscriptForm.tsx

```diff
@@ -86,20 +121,78 @@ const ManuscriptForm: React.FC<ManuscriptFormProps> = ({
     handleSubmit,
     control,
     formState: { isSubmitting },
+    getValues,
     watch,
     setValue,
+    reset,
   } = methods;
 
   const watchType = watch('versions.0.type');
+  const watchLifecycle = watch('versions.0.lifecycle');
 
   useEffect(() => {
     if (!watchType) {
       setValue('versions.0.lifecycle', '');
     }
   }, [setValue, watchType]);
 
+  useEffect(() => {
```

Comment:
can you extract some functions from this? As is it's a bit hard to read, being extracted/split into named functions should make it easier to understand.